src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java

```diff
@@ -399,8 +404,110 @@ void getHearingLocations() {
+            .containsExactlyInAnyOrder(tuple("219164", COURT));
+    }
+
+    @DisplayName("When a case has been adjourned and the same venue has been selected, return the same venue")
+    @Test
+    void getHearingLocationsAdjournmentSameVenue() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        caseData.getAppeal().setHearingSubtype(HearingSubtype.builder()
+                .wantsHearingTypeTelephone("yes")
+                .hearingTelephoneNumber("07483871426").build());
+        caseData.getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("yes").build());
+
+        caseData.setHearings(Collections.singletonList(Hearing.builder()
+                    .value(uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails.builder()
+                    .venueId(String.valueOf("219164")).build())
+                .build()));
+
+        DynamicListItem item = new DynamicListItem("219164", "");
+        DynamicList list = new DynamicList(item, Collections.emptyList());
+
+        caseData.setAdjournCaseNextHearingVenue("sameVenue");
+        caseData.setAdjournCaseNextHearingVenueSelected(list);
+
+        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder);
+
+        assertThat(result)
+            .hasSize(1)
+            .extracting("locationId","locationType")
+            .containsExactlyInAnyOrder(tuple("219164", COURT));
+    }
+
+    @DisplayName("When a case has been adjourned but the next hearing is paper, return the override hearing locations")
+    @Test
+    void getHearingLocationsAdjournmentNewVenuePaperCase() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        DynamicListItem item = new DynamicListItem("219164", "");
+        DynamicList list = new DynamicList(item, Collections.emptyList());
+
+        caseData.setAdjournCaseNextHearingVenue("somewhereElse");
+        caseData.setAdjournCaseNextHearingVenueSelected(list);
+
+        List<HearingLocation> result = HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder);
+
+        assertThat(result)
+            .hasSize(2)
+            .extracting("locationId","locationType")
+            .containsExactlyInAnyOrder(
+                tuple("219164", COURT),
+                tuple("436578", COURT));
+    }
+
+    @DisplayName("If the adjournCaseNextHearingVenue isn't recognised, throw an error")
+    @Test
+    void getHearingLocationsFailOnWrongNextHearingVenue() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        caseData.getAppeal().setHearingSubtype(HearingSubtype.builder()
+                .wantsHearingTypeTelephone("yes")
+                .hearingTelephoneNumber("07483871426").build());
+        caseData.getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("yes").build());
+
+        caseData.setAdjournCaseNextHearingVenue("UnknownValue");
+
+        assertThrows(IllegalStateException.class,
```

Comment:
I think assertThatThrownBy is a better method to use here. Then you can chain .isInstanceOf() and .hasMessageContaining() on the end.

Reply:
Done